deploy/Dockerfile.local

```diff
@@ -0,0 +1,26 @@
+FROM golang:alpine as builder
```

Comment:
If we're adding a Dockerfile.local, should we add back the makefile recipe for it?

Reply:
Do we want a Dockerfile.local? I thought it was supposed to be here, but can delete it.

Reply:
I use it pretty often for e2e validation locally https://github.com/xmidt-org/argus/blob/v0.3.9/Makefile#L58 
I don't think a dedicated .local file is necessary though.